tests/compile-and-dump/Singletons/StandaloneDeriving.ghc82.template

```diff
@@ -0,0 +1,448 @@
+Singletons/StandaloneDeriving.hs:(0,0)-(0,0): Splicing declarations
+    singletons
+      [d| infixl 6 :*:
+          
+          data T a b = a :*: b
+          data S = S1 | S2 |]
+  ======>
+    infixl 6 :*:
+    data T a b = a :*: b
+    data S = S1 | S2
+    type (:*:$$$) (t :: a0123456789876543210) (t :: b0123456789876543210) =
+        (:*:) t t
+    instance SuppressUnusedWarnings (:*:$$) where
+      suppressUnusedWarnings
+        = snd ((GHC.Tuple.(,) (:*:$$###)) GHC.Tuple.())
+    data (:*:$$) (l :: a0123456789876543210) (l :: TyFun b0123456789876543210 (T a0123456789876543210 b0123456789876543210))
+      = forall arg. SameKind (Apply ((:*:$$) l) arg) ((:*:$$$) l arg) =>
+        (:*:$$###)
+    type instance Apply ((:*:$$) l) l = (:*:) l l
+    instance SuppressUnusedWarnings (:*:$) where
+      suppressUnusedWarnings
+        = snd ((GHC.Tuple.(,) (:*:$###)) GHC.Tuple.())
+    data (:*:$) (l :: TyFun a0123456789876543210 (TyFun b0123456789876543210 (T a0123456789876543210 b0123456789876543210)
+                                                  -> GHC.Types.Type))
+      = forall arg. SameKind (Apply (:*:$) arg) ((:*:$$) arg) =>
+        (:*:$###)
+    type instance Apply (:*:$) l = (:*:$$) l
+    type S1Sym0 = S1
+    type S2Sym0 = S2
+    infixl 6 :%*:
+    data instance Sing (z :: T a b)
+      = forall (n :: a) (n :: b). z ~ (:*:) n n =>
+        (Sing (n :: a)) :%*: (Sing (n :: b))
+    type ST = (Sing :: T a b -> GHC.Types.Type)
+    instance (SingKind a, SingKind b) => SingKind (T a b) where
+      type Demote (T a b) = T (Demote a) (Demote b)
+      fromSing ((:%*:) b b) = ((:*:) (fromSing b)) (fromSing b)
+      toSing ((:*:) b b)
+        = case
+              (GHC.Tuple.(,) (toSing b :: SomeSing a)) (toSing b :: SomeSing b)
+          of {
+            GHC.Tuple.(,) (SomeSing c) (SomeSing c)
+              -> SomeSing (((:%*:) c) c) }
+    data instance Sing (z :: S) = z ~ S1 => SS1 | z ~ S2 => SS2
+    type SS = (Sing :: S -> GHC.Types.Type)
+    instance SingKind S where
+      type Demote S = S
+      fromSing SS1 = S1
+      fromSing SS2 = S2
+      toSing S1 = SomeSing SS1
+      toSing S2 = SomeSing SS2
+    instance (SingI n, SingI n) =>
+             SingI ((:*:) (n :: a) (n :: b)) where
+      sing = ((:%*:) sing) sing
+    instance SingI S1 where
+      sing = SS1
+    instance SingI S2 where
+      sing = SS2
+Singletons/StandaloneDeriving.hs:(0,0)-(0,0): Splicing declarations
+    singletons
+      [d| deriving instance Enum S
+          deriving instance Bounded S
+          deriving instance Show S
+          deriving instance Ord S
+          deriving instance Eq S
+          deriving instance Show a => Show (T a ())
+          deriving instance Ord a => Ord (T a ())
+          deriving instance Eq a => Eq (T a ()) |]
```

Comment:
Huh, that's not the order in which I originally wrote those! https://ghc.haskell.org/trac/ghc/ticket/13799